Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,5 +1,69 @@
-function setAlarm() {}
+function formatTimeUnit(unit) {
+  return unit < 10 ? `0${unit}` : unit;
+}
+
+const heading = document.querySelector("#timeRemaining");
+const stopBtn = document.querySelector("#stop");
+heading.innerText = "Time Remaining: 00:00";
+const headerText = 'Remaining Time:'
+
+//===========setAlarm======================
+function setAlarm() {
+  const totalTimeInSeconds = parseInt(
+    document.querySelector("#alarmSet").value
+  );
+  setInitialMinutesAndSeconds(totalTimeInSeconds);
+  countDownEverySeconds(totalTimeInSeconds);
+}
+
+//=========formatAndDisplayTime=========================
+function formatAndDisplayTime(text, minutes, seconds ){
+  const formattedMinutes = formatTimeUnit(minutes)
+  const formattedSeconds = formatTimeUnit(seconds)
+  heading.innerText = `${text} ${formattedMinutes}:${formattedSeconds}`
+}
+
+//===============calculateMinutesAndSeconds================
+function calculateMinutesAndSeconds(totalTimeInSeconds){
```

Comment:
This works (and reduces duplication of logic), but you're still ending up with both of your callers doing exactly the same thing - calling this function and then passing it to `formatAndDisplayTime`.

Is there a reason `formatAndDisplayTime` couldn't accept `totalTimeInSecond`, and do the calculation of minutes+seconds itself?

Reply:
formatAndDisplayTime can do it. trying to fix it now